### PR TITLE
Create parent class for all MatrixWorker

### PR DESCRIPTION
**Pull request by bmarty**
Trying to normalize the way we use worker, and also try to catch all possible Exception to avoid breaking worker chains.

Parsing Exception is dealt in @ganfra 's PR #2129 

This is really dirty, I cannot to parsing and buildErrorResult() in the parent class, so many duplicated code... Not very proud of it.

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/content/UploadContentWorker.kt [hunk 1]
```diff
@@ -217,14 +213,7 @@ internal class UploadContentWorker(val context: Context, params: WorkerParameter
             }
         } catch (e: Exception) {
             Timber.e(e, "## FileService: ERROR")
-            notifyTracker(params) { contentUploadStateTracker.setFailure(it, e) }
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/content/UploadContentWorker.kt:221]: This line has disappeared, is it OK?

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/worker/SessionSafeCoroutineWorker.kt [hunk 2]
```diff
@@ -42,43 +43,44 @@ internal abstract class SessionSafeCoroutineWorker<PARAM : SessionWorkerParams>(
...
         return try {
-            doSafeWork(sessionComponent, params)
+            val sessionComponent = getSessionComponent(params.sessionId)
+                    ?: return buildErrorResult(params, "No session")
+
+            // Make sure to inject before handling error as you may need some dependencies to process them.
+            injectWith(sessionComponent)
+            if (params.lastFailureMessage != null) {
+                // Forward error to the next workers
+                return doOnError(params)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/worker/SessionSafeCoroutineWorker.kt:58]: maybe remove `return` statement here and use a `else` block

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/worker/SessionSafeCoroutineWorker.kt [hunk 3]
```diff
@@ -87,12 +89,12 @@ internal abstract class SessionSafeCoroutineWorker<PARAM : SessionWorkerParams>(
     open fun doOnError(params: PARAM): Result {
         // Forward the error
         return Result.success(inputData)
-                .also { Timber.e("Work cancelled due to input error from parent") }
+                .also { Timber.e("Work cancelled due to input error from parent: $params") }
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/worker/SessionSafeCoroutineWorker.kt:92]: it may log private stuff, no?

Comment (ganfra) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/worker/SessionSafeCoroutineWorker.kt:92]: Yup sorry was for debugging

**Review by BillCarsonFr — APPROVED** [review]

**Pull request merged**